src/renderers/webgl/WebGLRenderStates.js

```diff
@@ -67,10 +67,30 @@ function WebGLRenderState() {
 function WebGLRenderStates() {
 
 	var renderStates = {};
+	var hashCache = {};
 
 	function get( scene, camera ) {
 
-		var hash = scene.id + ',' + camera.id;
+		var hash;
+
+		if ( hashCache[ scene.id ] ) {
+
+			hash = hashCache[ scene.id ][ camera.id ];
+
+			if ( hash === undefined ) {
+
+				hash = scene.id + ',' + camera.id;
+				hashCache[ scene.id ][ camera.id ] = hash;
+
+			}
+
+		} else {
+
+			hash = scene.id + ',' + camera.id;
+			hashCache[ scene.id ] = {};
+			hashCache[ scene.id ][ camera.id ] = hash;
+
+		}
```

Comment:
I think we don't need to use hash with similar approach.

    function get( scene, camera ) {

        if ( renderStates[ scene.id ] === undefined ) {

            renderStates[ scene.id ] = {};
            renderStates[ scene.id ][ camera.id ] = new WebGLRenderState();

        } else {

            if ( renderStates[ scene.id ][ camera.id ] === undefined ) {

                renderStates[ scene.id ][ camera.id ] = new WebGLRenderState();

            }

        }

        return renderStates[ scene.id ][ camera.id ];

    }

If we use `Map`, we don't even need `.id` like `renderStates.get( scene ).get( camera );`.

Reply:
Map is now widely supported? I'm in favor of using it here.